From f2126d551bdbfc0913ad681ef9a6143e2194a899 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Sat, 5 Dec 2020 20:40:26 +0530
Subject: [PATCH 09/40] Revert "ANDROID: kbuild: don't preprocess
 module-lto.lds"

This reverts commit ef3feb510a466e2dc3f60a33a4b6ebc11bc4b670.
---
 Makefile                                     | 2 +-
 scripts/Makefile                             | 2 ++
 scripts/{module-lto.lds => module-lto.lds.S} | 6 ++++--
 3 files changed, 7 insertions(+), 3 deletions(-)
 rename scripts/{module-lto.lds => module-lto.lds.S} (88%)

diff --git a/Makefile b/Makefile
index 128d3bbd6789..c748d8027ed7 100644
--- a/Makefile
+++ b/Makefile
@@ -896,7 +896,7 @@ LD_FLAGS_LTO_CLANG := -mllvm -import-instr-limit=5
 KBUILD_LDFLAGS += $(LD_FLAGS_LTO_CLANG)
 KBUILD_LDFLAGS_MODULE += $(LD_FLAGS_LTO_CLANG)
 
-KBUILD_LDS_MODULE += $(srctree)/scripts/module-lto.lds
+KBUILD_LDS_MODULE += scripts/module-lto.lds
 endif
 
 ifdef CONFIG_LTO
diff --git a/scripts/Makefile b/scripts/Makefile
index 3e86b300f5a1..543603e3a789 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -31,6 +31,8 @@ always		:= $(hostprogs-y) $(hostprogs-m)
 # The following hostprogs-y programs are only build on demand
 hostprogs-y += unifdef
 
+extra-$(CONFIG_LTO_CLANG)   += module-lto.lds
+
 subdir-$(CONFIG_GCC_PLUGINS) += gcc-plugins
 subdir-$(CONFIG_MODVERSIONS) += genksyms
 subdir-$(CONFIG_SECURITY_SELINUX) += selinux
diff --git a/scripts/module-lto.lds b/scripts/module-lto.lds.S
similarity index 88%
rename from scripts/module-lto.lds
rename to scripts/module-lto.lds.S
index 89d3b1636feb..c0f4fdeb84a0 100644
--- a/scripts/module-lto.lds
+++ b/scripts/module-lto.lds.S
@@ -1,4 +1,6 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+#include <asm/page.h>
+
 /*
  * With CONFIG_LTO_CLANG, LLD always enables -fdata-sections and
  * -ffunction-sections, which increases the size of the final module.
@@ -22,9 +24,9 @@ SECTIONS {
 
 	/*
 	 * With CFI_CLANG, ensure __cfi_check is at the beginning of the
-	 * .text section, and that the section is aligned to 4k.
+	 * .text section, and that the section is aligned to page size.
 	 */
-	.text : ALIGN(4096) {
+	.text : ALIGN(PAGE_SIZE) {
 		*(.text.__cfi_check)
 		*(.text .text.[0-9a-zA-Z_]* .text..L.cfi*)
 	}
-- 
2.17.1

